File: week1/GamGomYang/BOJ1935.py
from typing import Sequence, List, Any

number = int(input())

sik = input()

sik_list = [0] * number

for i in range(number):
    sik_list[i] = int(input())   

stack = []

for i in sik:
    if 'A' <= i <= 'Z':
        stack.append(sik_list[ord(i) - ord('A')])
    else:
        str2 = stack.pop()
        str1 = stack.pop()

        if i == '+':
            stack.append(str1 + str2)
        elif i == '-':
            stack.append(str1 - str2)
        elif i == '/':
            stack.append(str1 / str2)
        elif i == '*':
            stack.append(str1 * str2)

print(f'{stack[0]:.2f}')

File: week1/GamGomYang/BOJ1935.java
import java.util.Scanner;
import java.util.Stack;

public class PostfixCalculator {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int number = scanner.nextInt();
        scanner.nextLine();

        String sik = scanner.nextLine();

        int[] sikList = new int[number];

        for (int i = 0; i < number; i++) {
            sikList[i] = scanner.nextInt();
        }

        Stack<Double> stack = new Stack<>();

        for (char ch : sik.toCharArray()) {
            if ('A' <= ch && ch <= 'Z') {
                stack.push((double) sikList[ch - 'A']);
            } else {
                double str2 = stack.pop();
                double str1 = stack.pop();

                switch (ch) {
                    case '+':
                        stack.push(str1 + str2);
                        break;
                    case '-':
                        stack.push(str1 - str2);
                        break;
                    case '/':
                        stack.push(str1 / str2);
                        break;
                    case '*':
                        stack.push(str1 * str2);
                        break;
                }
            }
        }

        System.out.printf("%.2f", stack.pop());
    }
}
